components/sync_manager/src/syncmanager.udl

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+namespace syncmanager { };
+
+[Error]
+enum SyncManagerError {
+    "UnknownEngine",
+    "UnsupportedFeature",
+    "Sync15Error",
+    "UrlParseError",
+    "InterruptedError",
+    "JsonError",
+    "LoginsError",
+    "PlacesError",
+    "AnyhowError",
+};
+
+dictionary SyncParams {
+    // Why are we performing this sync?
+    SyncReason reason;
+    // Which engines should we sync?
+    SyncEngineSelection engines;
+    // TODO
+    record<DOMString, boolean> enabled_changes;
+    // Keys to encrypt/decrypt data from local database files.  These are
+    // separate from the key we use to encrypt the sync payload as a whole.
+    record<DOMString, string> local_encryption_keys;
+    // Authorization for the sync server
+    SyncAuthInfo auth_info;
+    // TODO
+    string? persisted_state;
+    // TODO
+    DeviceSettings device_settings;
+};
+
+[Enum]
+interface SyncEngineSelection {
+    All();
+    Some(sequence<string> engines);
+};
+
+enum SyncReason {
+    "Scheduled",
+    "User",
+    "PreSleep",
+    "Startup",
+    "EnabledChange",
+};
+
+dictionary SyncAuthInfo {
+    string kid;
+    string fxa_access_token;
+    string sync_key;
+    string tokenserver_url;
+};
+
+dictionary DeviceSettings {
+    string fxa_device_id;
+    string name;
+    DeviceType kind;
+};
+
+enum DeviceType {
+    "Desktop",
+    "Mobile",
+    "Tablet",
+    "VR",
+    "TV",
+};
+
+dictionary SyncResult {
+    // Result from the sync server
+    ServiceStatus status;
+    // Engines that synced successfully
+    sequence<string> successful;
+    // Maps the names of engines that failed to sync to the reason why
+    record<DOMString, string> failures;
+    // TODO
+    string persisted_state;
+    // TODO
+    sequence<string>? declined;
```

Comment:
The list of engines which are marked as "declined" (ie, disabled) on the sync server. The list of declined engines is global to the account rather than to the device. Apps should use this after every sync to update the local state (ie, to ensure that their Sync UI correctly reflects what engines are enabled and disabled), because these could change after every sync.